dcapal-frontend/src/components/allocationFlow/steps/invest.js

```diff
@@ -12,22 +12,20 @@ import classNames from "classnames";
 import { Trans, useTranslation } from "react-i18next";
 import { spawn, Thread, Worker } from "threads";
 import { replacer } from "@utils/index.js";
+
 const amtDecimals = 2;
 
-const buildProblemInput = (assets, useWholeShares) => {
+const buildProblemInput = (assets) => {
   return Object.values(assets).reduce(
     (as, a) => ({
       ...as,
       [a.symbol]: {
-        // Common input
         symbol: a.symbol,
         target_weight: a.targetWeight / 100,
-        // Use whole shares input
-        ...(useWholeShares && { shares: a.qty }),
-        ...(useWholeShares && { price: a.price }),
-        ...(useWholeShares && { is_whole_shares: isWholeShares(a.aclass) }),
-        // Use partial shares input
-        ...(!useWholeShares && { current_amount: a.amount }),
+        shares: a.qty,
+        price: a.price,
+        is_whole_shares: a.is_whole_shares,
```

Comment:
Are you sure about this? IIRC, there's no `is_whole_shares` field in the `asset` object. The way you implemented it before was right:
```suggestion
        is_whole_shares: isWholeShares(a.aclass),
```